src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -2484,7 +2491,90 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  '/nodes/{nodeId}/calculateSize':
+    post:
+      tags:
+        - nodes
+      summary: Calculate a folder size
+      description: |
+        
+        This may be executed against a node specified by **nodeId** and this nodeId must be a valid node ID.
+        Furthermore, this POST endpoint will be executed **asynchronously** with a `202` HTTP response signifying that
+        the request has been accepted successfully.
+        
+        The response body will include the ```nodeId``` of the pending action, which can be used in a
+        **GET/calculateSize** endpoint to check if the action's status has been completed, at which point the result will be returned, comprising ```the size of the node in bytes```.
+        
+        Essentially, ```maxItems``` ***parameter*** is utilized for pagination purposes.
+        
+        By default, **results are limited to the first 100.**
+        Results can be restricted using "paging". For example:
+        
+        ```JSON
+        "paging": {
+          "maxItems": "50",
+          "skipCount": "28"
+        }
+        ```
+        This assists in handling scenarios where large files or subfolders exist within a folder, allowing the folder size determination mechanism to function in chunks.
 
+      operationId: calculateSize
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/nodeTargetIdParam'
+        - $ref: '#/parameters/maxItemsForSizeParam'
+      responses:
+        '202':
+          description: Request Accepted
+          schema:
+            $ref: '#/definitions/sizeAcknowledgedResponse'
+        '401':
+          description: Authentication failed
+        '403':
+          description: Current user does not have permission for **nodeId**
+        '404':
+          description: Entity with **nodeId** does not exist
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
```

Comment:
What error code are we returning in case its a valid node id but this is a retention-schedule node id

Reply:
We are returning **404- nodeId does not exist**, as we are only considering folder type node.

Reply:
But here node id exist in the system but it has different nose type. I think we should have a different message and may be status code as well. Please check other endpoints.

Reply:
Updated with **400 status code.** with description

Reply:
Plz once validate the message with other endpoints in the swagger so that we are having similar standards. Rest looks good to me.